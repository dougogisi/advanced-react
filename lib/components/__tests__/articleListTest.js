import React from 'react';
import ArticleList from '../ArticleList';

// import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const testProps = {
  articles: {
    a: { id: 'a', title: '', body: '' },
    b: { id: 'b', title: '', body: '' },
  }
};

describe('ArticleList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );

    expect(wrapper.find('ArticleContainer').length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
