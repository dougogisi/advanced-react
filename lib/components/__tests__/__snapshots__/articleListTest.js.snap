// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div>
      <div
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <div
      style={
        Object {
          "color": "#888",
          "fontSize": "0.85em",
        }
      }
    >
      Invalid Date
    </div>
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div>
      <div
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      />
    </div>
  </div>
</div>
`;
